File: PAT (Advanced-Level)/C/1009.c
#include <stdio.h>

int main(int argc, char const *argv[]){
    int k, a;
    double input[1000+1] = {0.0};
    double result[2000+1] = {0.0};
    double b;

    scanf("%d", &k);
    for (int i = 0; i < k; i++){
        scanf("%d %lf", &a, &b);
        input[a] = b;
    }

    scanf("%d", &k);
    for (int i = 0; i < k; i++){
        scanf("%d %lf", &a, &b);
        for (int j = 0; j <= 1000; j++){
            result[j+a] += input[j]*b;
        }
    }
    int count = 0;
    for (int i = 0; i <= 2000; i++){
        if(result[i] != 0.0)
            count++;
    }
    printf("%d", count);

    for (int i = 2000; i >= 0; i--){
        if(result[i] != 0.0)
            printf(" %d %.1f", i, result[i]);
    }
    return 0;
}

File: PAT (Advanced-Level)/Python/1009.py
#!/usr/bin/python3
# -*- coding: utf-8 -*-

inputData = [0.0 for x in range(1000+1)]
result = [0.0 for x in range(2000+1)]

data = input().split()
for i in range(int(data[0])):
    a = int(data[2*i+1])
    b = float(data[2*i+2])
    inputData[a] = b

data = input().split()
for i in range(int(data[0])):
    a = int(data[2*i+1])
    b = float(data[2*i+2])
    for j in range(1001):
        result[j+a] += inputData[j]*b

count = 0
for i in result:
    if i != 0.0:
        count += 1
print(count, end = '')

for i in range(2000, -1, -1):
    if result[i] != 0.0:
        print(" %d %.1f" % (i, result[i]), end = '')

